internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,142 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
-// Runtime defines the ABI interface exposed to actors.
-type Runtime interface {
-	Message() *types.UnsignedMessage
-	Storage() Storage
-	Send(to address.Address, method types.MethodID, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
-	AddressForNewActor() (address.Address, error)
-	BlockHeight() *types.BlockHeight
-	MyBalance() types.AttoFIL
-	IsFromAccountActor() bool
-	Charge(cost types.GasUnits) error
-	SampleChainRandomness(sampleHeight *types.BlockHeight) ([]byte, error)
+// Runtime2 is the VM.
+type Runtime2 struct {
+	CurerntEpoch() Epoch
+	// Review: for @go-filecoin, should we turn the offset into a RandomnessEvent?
+	Randomness(epoch Epoch, offset uint64) Randomness
+	Send(input InvocInput) result.InvocOutput
+	// SharedStorage is an IPLD store shared by all actors.
+	// Review: for @go-filecoin, we can skip this until we have actor code that actually needs this
+	// Review: for @spec, Add expiration (Epoch's) on PUT? (the question is mostly for global, but the local store might want to advise on this too).
+	// Review: for @spec, how will gas accounting work on global? PUT fee, GET fee, OWNER Fee?
+	SharedStorage() IpldStorage2
+}
+
+// DispatchContext is the context the actor dispatcher receives.
+// Review: for @go-filecoin, do we want to have the `ctx.ValidateCaller(blahblah)` at the top of each method, or hidden away on the dispatch table?
+type DispatchContext interface {
+	// Review: for @go-filecoin, simplified the API to be just one method.
+	ValidateCaller(CallerPattern)
+	// InvocationContext is the context to be passed down the actor method.
+	InvocationContext() InvocationContext
+}
+
+// InvocationContext is passed to the actors on each method call..
+type InvocationContext interface {
+	// Review: for @go-filecoin, composition or inheritance?
+	Runtime() Runtime2
+	Caller() address.Address
+	StateHandler() ActorStateHandler
+
+	// Review: for @go-filecoin+@spec, renamed `ValueSupplied` to `ValueReceived`. The "supplied" is not clear by whom.
+	ValueReceived() types.AttoFIL
+	Balance() types.AttoFIL
+
+	LocalStorage() IpldStorage2
```

Comment:
Pending information to the contrary, let's drop this in favour of the `SharedStorage`. Even if that changes, I expect it to be resolved pretty quickly such that we don't have any migration friction to worry about.